frontend/src/metabase/ui/components/inputs/TimeInput/TimeInput.tsx

```diff
@@ -19,43 +19,55 @@ export function TimeInput({
   value,
   defaultValue = value,
   onChange,
+  onFocus,
+  onBlur,
   ...props
 }: TimeInputProps) {
-  const [inputValue, setInputValue] = useState(
-    defaultValue ? formatTime(defaultValue) : "",
-  );
+  const [inputValue, setInputValue] = useState(formatValue(defaultValue));
```

Comment:
Just wanted to make the implementation close to `NumberInput`, no behavior changes